core/utils/colour.js

```diff
@@ -92,7 +94,12 @@ Blockly.utils.colour.rgbToHex = function(r, g, b) {
  * @return {!Array.<number>} RGB representation of the colour.
  */
 Blockly.utils.colour.hexToRgb = function(hexColour) {
-  var rgb = parseInt(hexColour.substr(1), 16);
+  var hex = Blockly.utils.colour.parse(hexColour);
```

Comment:
Given this function converts a hex to rgb, I wouldn't want to pass it through colour.parse, instead we could do the following: 
```js
  hexColour = hexColour.substring(0, 2) == '0x' ?
    hexColour.substring(2) : hexColour.substring(1);
```
and then set rgb to:
```js
  var rgb = parseInt(hexColour, 16);
```

I would also update the @param to:
```js
 * @param {string} hexColour Colour in '#ff0000' or '0xff0000' format.
```

Reply:
hm, this means hex shorthand ("#fff") would parse incorrectly though. i guess the documentation could be updated to exclude that input format?

Reply:
Fair enough. Let's update the jsdoc and parameter name to accept a colour. (instead of hexColour), and pass it through the parse method as you're doing.